### ISSUE TITLE
exit + stay connected = endless loop

### ISSUE THREAD

**Issue by cinnamonsweets** [connectbot/connectbot#631]
I have a script that runs on connect that ends with exit. I accidentally tapped stay connected and didn't realize it until I ran the script. It ran the script continuously! I use connectbot for home automation and this could have catastrophic results.

Shouldn't it run only ONCE when the profile is first run and not again when "stay connected" reconnects automatically?

**Comment by kruton** [comment]
It's difficult to know what someone will run. Maybe it's a `tail -f` to read a log? If I disable being able to reconnect when a command is entered, someone else will complain.

**Comment by cinnamonsweets** [comment]
You are right, running only once per profile session wouldn't work well for monitoring. My concern is safeguarding against unintentional script looping that may cause damage in home automation. Perhaps require two steps to setup "stay connected"?

Relabel toggle "stay connected" to "reconnect on disconnect" for ease of association.

Add textbox labeled "reconnect count" that is disabled unless "reconnect on disconnect" is toggled on.
Textbox defaults to zero or empty and leaving it as such causes "reconnect on disconnect" to toggle off.
Negative number is infinite reconnects.
